src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -170,35 +170,45 @@ extern "C" int32_t CryptoNative_SslSessionReused(SSL* ssl)
     return SSL_session_reused(ssl) == 1;
 }
 
+static bool StringSpanEquals(const char* lhs, const char* rhs, size_t lhsLength)
+{
+    if (lhsLength != strlen(rhs))
+    {
+        return false;
+    }
+
+    return strncmp(lhs, rhs, lhsLength) == 0;
```

Comment:
[`strncmp`](http://en.cppreference.com/w/c/string/byte/strncmp) is particularly used to check up to a certain length. Can't we just use [`strcmp`](http://en.cppreference.com/w/c/string/byte/strcmp) instead to avoid the branch (and this whole function); since we want to compare the whole string?

```c
if(strcmp(encryption, "3DES(168)") == 0) //etc.
```